Scala2-Functional-Programming/Week6/forcomp/src/main/scala/forcomp/Anagrams.scala

```diff
@@ -134,7 +160,26 @@ object Anagrams extends AnagramsInterface {
    *
    *  Note: There is only one anagram of an empty sentence.
    */
-  def sentenceAnagrams(sentence: Sentence): List[Sentence] = ???
+  def sentenceAnagrams(sentence: Sentence): List[Sentence] = {
+    val lettersOcc: Occurrences = sentenceOccurrences(sentence)
+
+    def helper(current: Sentence, leftOccurrence: Occurrences): List[Sentence]  = {
```

Comment:
There is a simpler way to implement this, a way where this function accepts only one parameter.
Choose whether you want to invest time to implement it this way.
I encourage you to do so.